app/src/main/java/com/example/tic_tac_toe_assignment1/MainActivity.kt

```diff
@@ -1,32 +1,55 @@
 package com.example.tic_tac_toe_assignment1
 
 import android.os.Bundle
+import android.widget.Button
+import android.widget.GridLayout
+import android.widget.TextView
 import androidx.activity.ComponentActivity
-import androidx.activity.compose.setContent
-import androidx.activity.enableEdgeToEdge
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.padding
-import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.tooling.preview.Preview
 import com.example.tic_tac_toe_assignment1.ui.theme.Tictactoeassignment1Theme
 
 class MainActivity : ComponentActivity() {
+    private lateinit var gridLayout: GridLayout
+    private lateinit var statusText: TextView
+    private lateinit var playAgainButton: Button
+    private var buttons = arrayOfNulls<Button>(9)
+    private var xTurn = true
+    private var gameOver = false
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        enableEdgeToEdge()
-        setContent {
-            Tictactoeassignment1Theme {
-                Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
-                    Greeting(
-                        name = "Android",
-                        modifier = Modifier.padding(innerPadding)
-                    )
-                }
-            }
+        setContentView(R.layout.activity_main);
+
+        gridLayout = findViewById(R.id.gridLayout)
+        statusText = findViewById(R.id.statusText)
+        playAgainButton = findViewById(R.id.playAgainButton)
+
+        for (i in 0 until 9) {
```

Comment:
we dont do 9 as a magic number
we use gridLayout.childCount
so if something changein the future you won't need to remember changing here too

Reply:
u right